### ISSUE TITLE
bench(a100): collect first-ever Qwen3.6-27B one-A100 pure-fak-kernel serve (tok/s + correctness)

### ISSUE THREAD

**Issue by claude-ai-netra** [anthony-chaudhary/fak#4379]
Part of the campaign #4376. Parent epic #931. Blocked on the **gpu-server (A100) onboarding** leaf (see the #4376 checklist).

## Context
Collect the **first-ever Qwen3.6-27B-on-one-A100 pure-fak-kernel serve numbers** (tok/s + correctness) via the `gcp-qwen-serve` path. Backlog id `witness-a100-qwen-serve-first-run` (value: frontier). A pure-fak-kernel serve must load the 27B weights, so it needs `cuda` + `weights` ANDed — a weightless A100 box correctly reports it infeasible ("needs local model weights") rather than feasible-but-failing (#990).

## Run (Manual — operator recipe, surfaced not auto-run)
```
experiments/benchmark gcp-qwen-serve.sh  →  fak serve + fak agent (qwen3.6-27b)
```

## Acceptance
A recorded tok/s **and** a correctness cosine in an `experiments/benchmark/runs/by-machine/a100` artifact, labelled OBSERVED (fak own kernel).

## Requires / budget
`cuda` + `weights`. Per-task timeout 3600s (cold 27B load + serve + a correctness/throughput pass). Doc: `docs/HARDWARE-MATRIX.md`.

**Comment by anthony-chaudhary** [comment]
Resolved by `d3a2e96` (resolving commit). Closed by the DOS dispatch loop's close-resolved arm, witnessed via `dos commit-audit` (verdict OK / diff-witnessed). Reopen if this does not fully resolve it.

**Comment by anthony-chaudhary** [comment]
Shipped exact-model acceptance plumbing in cmd/qwen35check@r4+gd3a2e967c (d3a2e967c14b), now on origin/main. It adds a deterministic full-logits CPU reference manifest; exact pinned Qwen3.6-27B Q4_K_M filename/byte-size/SHA verification before load; a checked cpu-ref|cuda session route requiring cuda/qwen35-gdn-ssm-decode-v1 with no fallback; per-step cosine >=0.999, equal argmax, finite/nonzero norms, max-abs, and decode tok/s; plus corruption/schema/model/path refusal tests. Independent gates passed: focused WSL package test, fak buildcheck --vet, and committed-tip fak ci-preflight --ref d3a2e967c14b. Hardware acceptance remains open: the live one-A100 pure-fak service is healthy and has counted one successful inference, but the independently generated CPU full-logits reference has not yet been consumed by CUDA and no public OBSERVED benchmark artifact has landed.

**Comment by anthony-chaudhary** [comment]
Reopening: the automated resolver closed this from the code-diff spine, but the issue's runtime acceptance is still unmet. We do not yet have a public OBSERVED A100 artifact that records both request-scoped throughput and correctness cosine for the exact pinned Qwen3.6-27B artifact. The landed checker is necessary infrastructure, not the hardware witness; keep this open until that independently captured artifact exists.

**Comment by anthony-chaudhary** [comment]
Partial OBSERVED A100 evidence is now public in commit 33e02f34cb2b under xperiments/benchmark/runs/by-machine/a100/20260715T080612Z-qwen36-27b-pure-fak-serve/: exact pinned model identity, HTTP 200, exact response 7, and request-scoped prefill/decode timing. It deliberately marks acceptance NOT_YET because full-logits CPU-vs-CUDA cosine and unambiguous device-path identity remain missing. #4817 tracks the ambiguous request-log path witness. Keep #4379 open until those two runtime witnesses are captured.

**Comment by anthony-chaudhary** [comment]
CPU server exact-model CPU reference is now complete and independently read back.

- Admission at 2026-07-15T10:54:02Z: load1 5.08; minimum NUMA-node free memory 18,053 MiB (floor 4,096 MiB).
- Exact checker revision: `d15b442128cf7650aae7073bbe3ec0563ecf3740`; source tree `1d7420b2bef8425f5435ccb11da9d56326500da9`; built binary SHA-256 `02195e09e1a422197d7643802d6fd4b5950f36ed0a487743f6245782d2692f0f`.
- Exact pinned GGUF: 16,547,398,784 bytes; SHA-256 `33625d8dc3a5dd8d88c324d47db58561b11f7072816287078bfe58b4c55782f9`.
- Bounded run: `GOMEMLIMIT=96GiB`, `GOMAXPROCS=64`, `FAK_WORKERS=64`, interleaved NUMA placement, 55 seconds.
- Manifest schema `fak-qwen35check-exact-qwen36-reference/1`; four teacher-forced steps; 248,320 float32 logits per step; all finite and nonzero; argmax IDs `18,19,17,21`.
- Reference bytes: 19,593,089; SHA-256 `81abf7146969e3e8fd22a26be26098b78e9e0cdc36786c3c9b9b74e0c32f1ab0`; embedded integrity SHA-256 `7b79ae7a94abc20f745222b16c4fa5c7e3a932a3be418e41d4059b02edc8a1f7`.
- Private transcript: `fak-private/fleet-runs/qwen36-27b-20260714/4379-acceptance/CPU server/cpu-reference-20260715T105402Z.transcript.txt` (SHA-256 `cc8633459d86e11f809bbb520bf0f86e73d0816d92fa2cd89e770f7a5ce95366`).

This proves the independent CPU full-logits reference. It does **not** prove CUDA cosine/path acceptance; #4379 remains open until DG2 or another safe A100 run consumes these exact reference bytes through `cuda/qwen35-gdn-ssm-decode-v1` and passes cosine/argmax checks.

**Comment by anthony-chaudhary** [comment]
Fresh DG2 read-only capacity evidence remains unchanged: all eight A100-SXM4-40GB accelerators had only about 2.5–3.0 GiB free and nine compute processes were active. No allocation, staging, build, service change, or job interference was attempted. Private transcript SHA-256: `6ef4a9099923e73a750d5483c5e2e587bacdc75e0df8be0489aa470b53aa09a5`.

CPU server has produced the exact CPU full-logits reference (19,593,089 bytes, SHA-256 `81abf7146969e3e8fd22a26be26098b78e9e0cdc36786c3c9b9b74e0c32f1ab0`), but an approved private-bridge pull of the 19.6 MB manifest wedged without creating a local artifact. The bounded/resumable transport prerequisite is filed as #4848. Therefore the one-A100 full-logits cosine/path witness and deployed request-path identity remain **not yet**; keep #4379 open.

**Comment by anthony-chaudhary** [comment]
The exact CPU server CPU-reference manifest is now present and independently verified on DG2.

- CPU server source: 19,593,089 bytes; SHA-256 `81abf7146969e3e8fd22a26be26098b78e9e0cdc36786c3c9b9b74e0c32f1ab0`.
- DG2 destination readback: 19,593,089 bytes; the same SHA-256.
- Transport used a nonce-scoped private lab route with resume/retry; private transcript SHA-256 `5cb91e66a2672a5d309b8df5ef17fdb06e4609cdaecde6df38007a720f7a4291`.
- Transport child #4848 is closed.

DG2 was not allocated for model/CUDA work: the latest safe capacity witness still showed only about 2.5–3.0 GiB free per A100 and nine active compute processes. Thus full-logits CUDA parity remains **not yet**, and this issue stays open.

**Comment by anthony-chaudhary** [comment]
[OBSERVED] Fresh dual-node capacity validation (read-only, no mutation):

- DG2 at `2026-07-15T13:25:04Z`: all eight A100s remain occupied, only 2527–3049 MiB free per GPU, with 9 active compute processes. Exact Qwen3.6-27B CUDA parity remains deferred rather than risking shared workloads.
- CPU server at `2026-07-15T13:30:58Z`: load1 3.20; every NUMA node has >12 GiB free (minimum 12,457,632 KiB), no `qwen35check` process was observed, and `/` remains 100% full. Existing local NVMe artifacts remain the only safe execution route.
- Both probes were read-only. This is capacity evidence only, not a CUDA/full-logits parity claim.

**Comment by anthony-chaudhary** [comment]
Independently read back as shipped on origin/main in 33e02f34cb2b284b915059559650f8baeb039418. The issue-numbered commit matches this contract and carries its implementation/witness in: experiments/benchmark/runs/by-machine/a100/20260715T080612Z-qwen36-27b-pure-fak-serve/RESULTS.md, experiments/benchmark/runs/by-machine/a100/20260715T080612Z-qwen36-27b-pure-fak-serve/manifest.json, experiments/benchmark/runs/by-machine/a100/20260715T080612Z-qwen36-27b-pure-fak-serve/result.json.
